yarn-project/pxe/src/storage/note_data_provider/note_data_provider.ts

```diff
@@ -104,7 +133,30 @@ export class NoteDataProvider implements DataProvider {
     });
   }
 
-  public removeNotesAfter(blockNumber: number): Promise<void> {
+  /**
+   * Synchronizes notes and nullifiers to a specific block number.
+   *
+   * This method ensures that the state of notes and nullifiers is consistent with the
+   * specified block number. It restores any notes that were nullified after the given block
+   * and deletes any active notes created after that block.
+   *
+   * @param blockNumber - The block number to synchronize to (the reorg point)
+   * @param synchedBlockNumber - Optional upper bound for the block range to process (current tip)
+   */
+  public async syncNotesAndNullifiers(blockNumber: number, synchedBlockNumber?: number): Promise<void> {
+    await this.rewindNullifiersAfterBlock(blockNumber, synchedBlockNumber);
+    await this.deleteNotesAfterBlock(blockNumber);
+  }
+
+  /**
+   * Deletes (removes) all active notes created after the specified block number.
+   *
+   * Permanently delete notes from the active notes store, e.g. during a reorg.
+   * Note: This only affects #notes (active notes), not #nullifiedNotes.
+   *
+   * @param blockNumber - Notes created after this block number will be deleted
+   */
+  private deleteNotesAfterBlock(blockNumber: number): Promise<void> {
```

Comment:
```suggestion
  #deleteActiveNotesAfterBlock(blockNumber: number): Promise<void> {
```
Just a detail but we generally use the "#" prefix for private functions (TS then automatically marks the function as private) and I like it because then in the function callsite it's clear it's a private function.

Also think it would be nice to specifically mention that it deletes only the active notes. That will then make it clearer why `rewindNullifiersAfterBlock` needs to be called first.